zaraz.go

```diff
@@ -0,0 +1,234 @@
+	Result []ZarazConfigRow `json:"result"`
+	Response
+	ResultInfo `json:"result_info"`
+}
+
+type GetZarazConfigHistoryParams struct {
+	ResultInfo
+}
+
+type GetZarazConfigsByIdResponse = map[string]interface{}
+
+// listZarazConfigHistoryDefaultPageSize represents the default per_page size of the API.
+var listZarazConfigHistoryDefaultPageSize int = 100
+
+func (api *API) GetZarazConfig(ctx context.Context, rc *ResourceContainer) (ZarazConfigResponse, error) {
+	if rc.Identifier == "" {
+		return ZarazConfigResponse{}, ErrMissingZoneID
+	}
+
+	uri := fmt.Sprintf("/zones/%s/settings/zaraz/v2/config", rc.Identifier)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return ZarazConfigResponse{}, err
+	}
+
+	var recordResp ZarazConfigResponse
+	err = json.Unmarshal(res, &recordResp)
+	if err != nil {
+		return ZarazConfigResponse{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return recordResp, nil
+}
+
+func (api *API) UpdateZarazConfig(ctx context.Context, rc *ResourceContainer, params UpdateZarazConfigParams) (ZarazConfigResponse, error) {
+	if rc.Identifier == "" {
+		return ZarazConfigResponse{}, ErrMissingZoneID
+	}
+
+	uri := fmt.Sprintf("/zones/%s/settings/zaraz/v2/config", rc.Identifier)
+	res, err := api.makeRequestContext(ctx, http.MethodPut, uri, params)
+	if err != nil {
+		return ZarazConfigResponse{}, err
+	}
+
+	var updateResp ZarazConfigResponse
+	err = json.Unmarshal(res, &updateResp)
+	if err != nil {
+		return ZarazConfigResponse{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return updateResp, nil
+}
+
+func (api *API) GetZarazWorkflow(ctx context.Context, rc *ResourceContainer) (ZarazWorkflowResponse, error) {
+	if rc.Identifier == "" {
+		return ZarazWorkflowResponse{}, ErrMissingZoneID
+	}
+
+	uri := fmt.Sprintf("/zones/%s/settings/zaraz/v2/workflow", rc.Identifier)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return ZarazWorkflowResponse{}, err
+	}
+
+	var response ZarazWorkflowResponse
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		return ZarazWorkflowResponse{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return response, nil
+}
+
+func (api *API) UpdateZarazWorkflow(ctx context.Context, rc *ResourceContainer, workflowToUpdate string) (ZarazWorkflowResponse, error) {
+	if rc.Identifier == "" {
+		return ZarazWorkflowResponse{}, ErrMissingZoneID
+	}
+
+	uri := fmt.Sprintf("/zones/%s/settings/zaraz/v2/workflow", rc.Identifier)
+	res, err := api.makeRequestContext(ctx, http.MethodPut, uri, workflowToUpdate)
+	if err != nil {
+		return ZarazWorkflowResponse{}, err
+	}
+
+	var response ZarazWorkflowResponse
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		return ZarazWorkflowResponse{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return response, nil
+}
+
+func (api *API) PublishZarazConfig(ctx context.Context, rc *ResourceContainer, description string) (ZarazPublishResponse, error) {
```

Comment:
the third parameter of all methods should be the params - see https://github.com/cloudflare/cloudflare-go/blob/master/docs/conventions.md#methods for examples.